src/app/workspace/resource/resource-instance-form/select-properties/select-properties.component.ts

```diff
@@ -45,13 +47,18 @@ export class SelectPropertiesComponent implements OnInit {
                     // each property will also have a filtered array to be used when deleting a value.
                     // see the deleteValue method below for more info
                     this.propertyValuesKeyValuePair[prop.id + '-filtered'] = [0];
+
+                     // check the cardinality to know if the prop is required or not
+                     this.isPropRequired(prop.id);
+                     this.propertyValuesKeyValuePair[prop.id + '-cardinality'] = [this.isRequiredProp ? 1 : 0];
```

Comment:
Can you add a comment explaining why we are adding this to the KeyValuePair?

Reply:
done in f131477